google-cloud-no-speech/src/intents/handlers/NoInputIntentHandler.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const { NO_INPUT_EVENT_NAMES } = require('../GenesysNoInputEventNames');
+const { NO_INPUT_COUNT, NO_INPUT_LIMIT } = require('../GenesysParameterNames');
+
+class NoInputIntentHandler {
+    constructor() {
+    }
+
+    canHandle(request) {
+        let retVal = false;
+
+        // Allow this handler to run if the queryText on the queryResult matches any of the
+        // Genesys-defined no input event names.
+        if (request.body && request.body.queryResult && request.body.queryResult.queryText) {
+            retVal = NO_INPUT_EVENT_NAMES.includes(request.body.queryResult.queryText);
+        }
+
+        return retVal;
+    }
+
+    handle(request, response) {
+        if (request.body && request.body.originalDetectIntentRequest && request.body.originalDetectIntentRequest.payload) {
+            const originalDetectIntentRequestPayload = request.body.originalDetectIntentRequest.payload;
+            const noInputCount = originalDetectIntentRequestPayload[NO_INPUT_COUNT];
+            const noInputLimit = originalDetectIntentRequestPayload[NO_INPUT_LIMIT];
+            let newFulfillmentText = request.body.queryResult

```

Comment:
We might not need to initialize newFulfillmentText since if/else part is going to override this value.

Reply:
True.  I put it there as a fallback in case someone modifies the code though, so they at least can keep the original text.